File: replay.js
const fs = require('fs');
const log = fs.readFileSync('/Users/xProject/.gemini/antigravity/brain/bfdf87e1-c40d-40a3-9a1a-391208bf725a/.system_generated/logs/overview.txt', 'utf8');

let content = fs.readFileSync('/Users/xProject/Desktop/Startups/Dev Stuff (Startups)/Lbricol.ma/src/features/hosts/components/PropertySetupWizard.tsx', 'utf8');

const lines = log.trim().split('\n');
let operations = [];

for (const line of lines) {
    try {
        const obj = JSON.parse(line);
        if (obj.type === 'TOOL_CALL' || obj.type === 'PLANNER_RESPONSE') {
            if (obj.tool_calls) {
                for (const tc of obj.tool_calls) {
                    if (tc.name === 'multi_replace_file_content' || tc.name === 'replace_file_content') {
                        if (tc.args.TargetFile && tc.args.TargetFile.includes('PropertySetupWizard.tsx')) {
                            operations.push(tc.args);
                        }
                    }
                }
            }
        }
    } catch (e) {}
}

console.log('Found', operations.length, 'operations.');

let successCount = 0;
let failCount = 0;

for (let i = 0; i < operations.length; i++) {
    const op = operations[i];
    let chunks = op.ReplacementChunks;
    if (!chunks && op.TargetContent) {
        chunks = [op];
    }
    
    if (typeof chunks === 'string') {
        try {
            // Fix unescaped newlines in JSON string
            chunks = JSON.parse(chunks.replace(/\n/g, '\\n').replace(/\r/g, '\\r').replace(/\t/g, '\\t'));
        } catch(e) {
            try {
                // If strict parse fails, use Function evaluation
                chunks = new Function('return ' + chunks)();
            } catch(e2) {
                console.log('Failed to parse chunks for op', i);
                continue;
            }
        }
    }
    
    for (const chunk of chunks) {
        if (content.includes(chunk.TargetContent)) {
            content = content.replace(chunk.TargetContent, chunk.ReplacementContent);
            successCount++;
        } else {
            console.log('Failed to match chunk in op', i, 'target starting with:', chunk.TargetContent.substring(0, 30).replace(/\n/g, '\\n'));
            failCount++;
        }
    }
}

fs.writeFileSync('/Users/xProject/Desktop/Startups/Dev Stuff (Startups)/Lbricol.ma/src/features/hosts/components/PropertySetupWizard.tsx', content);
console.log('Applied diffs. Success:', successCount, 'Failures:', failCount);

File: replay.py
import json

log_path = '/Users/xProject/.gemini/antigravity/brain/bfdf87e1-c40d-40a3-9a1a-391208bf725a/.system_generated/logs/overview.txt'
target_file = '/Users/xProject/Desktop/Startups/Dev Stuff (Startups)/Lbricol.ma/src/features/hosts/components/PropertySetupWizard.tsx'

with open(log_path, 'r') as f:
    lines = f.readlines()

with open(target_file, 'r') as f:
    content = f.read()

operations = []

for line in lines:
    line = line.strip()
    if not line:
        continue
    try:
        obj = json.loads(line)
        if obj.get('type') in ['TOOL_CALL', 'PLANNER_RESPONSE']:
            for tc in obj.get('tool_calls', []):
                if tc.get('name') in ['multi_replace_file_content', 'replace_file_content']:
                    if 'PropertySetupWizard.tsx' in tc.get('args', {}).get('TargetFile', ''):
                        operations.append(tc['args'])
    except Exception as e:
        pass

print(f"Found {len(operations)} operations.")

success_count = 0
fail_count = 0

for i, op in enumerate(operations):
    chunks = op.get('ReplacementChunks')
    if not chunks and op.get('TargetContent'):
        chunks = [op]
    
    if isinstance(chunks, str):
        try:
            # The string might be JSON
            # Replace actual newlines with \n for json.loads
            chunks_str = chunks.replace('\n', '\\n').replace('\r', '\\r').replace('\t', '\\t')
            chunks = json.loads(chunks_str)
        except Exception as e:
            print(f"Failed to parse chunks for op {i}")
            continue
            
    for chunk in chunks:
        target = chunk.get('TargetContent', '')
        replacement = chunk.get('ReplacementContent', '')
        if target in content:
            content = content.replace(target, replacement)
            success_count += 1
        else:
            print(f"Failed to match chunk in op {i}")
            fail_count += 1

with open(target_file, 'w') as f:
    f.write(content)

print(f"Applied diffs. Success: {success_count}, Failures: {fail_count}")
